compose/src/main/kotlin/com/yourssu/handy/compose/foundation/SemanticColors.kt

```diff
@@ -0,0 +1,124 @@
+    val lineStatusPositive: Color = ColorViolet500,
+
+    // Button / Box / Primary
+    val buttonBoxPrimaryEnabled: Color = ColorViolet500,
+    val buttonBoxPrimaryPressed: Color = ColorViolet700,
+    val buttonBoxPrimaryDisabled: Color = ColorGray100,
+
+    // Button / Box / Secondary
+    val buttonBoxSecondaryEnabled: Color = ColorViolet50,
+    val buttonBoxSecondaryPressed: Color = ColorViolet200,
+    val buttonBoxSecondaryDisabled: Color = ColorGray100,
+
+    // Button / Box / Tertiary
+    val buttonBoxTertiaryEnabled: Color = ColorNeutralTransparent,
+    val buttonBoxTertiaryPressed: Color = ColorGray100,
+    val buttonBoxTertiaryDisabled: Color = ColorNeutralTransparent,
+
+    // Button / Text / Primary
+    val buttonTextPrimaryEnabled: Color = ColorNeutralTransparent,
+    val buttonTextPrimaryPressed: Color = ColorViolet50,
+    val buttonTextPrimaryDisabled: Color = ColorNeutralTransparent,
+
+    // Button / Text / Secondary
+    val buttonTextSecondaryEnabled: Color = ColorNeutralTransparent,
+    val buttonTextSecondaryPressed: Color = ColorGray100,
+    val buttonTextSecondaryDisabled: Color = ColorNeutralTransparent,
+
+    // Button / Radio
+    val buttonRadioSelected: Color = ColorViolet500,
+    val buttonRadioUnselected: Color = ColorGray200,
+    val buttonRadioDisabled: Color = ColorNeutralWhite,
+
+    // Fab / Primary
+    val buttonFabPrimaryEnabled: Color = ColorViolet500,
+    val buttonFabPrimaryPressed: Color = ColorViolet700,
+    val buttonFabPrimaryDisabled: Color = ColorGray100,
+
+    // Fab / Secondary
+    val buttonFabSecondaryEnabled: Color = ColorNeutralWhite,
+    val buttonFabSecondaryPressed: Color = ColorGray100,
+    val buttonFabSecondaryDisabled: Color = ColorNeutralWhite,
+
+    // Icon / Basic
+    val iconBasicPrimary: Color = ColorNeutralBlack,
+    val iconBasicSecondary: Color = ColorGray700,
+    val iconBasicTertiary: Color = ColorGray500,
+    val iconBasicDisabled: Color = ColorGray200,
+    val iconBasicWhite: Color = ColorNeutralWhite,
+
+    // Icon / Brand
+    val iconBrandPrimary: Color = ColorViolet500,
+    val iconBrandSecondary: Color = ColorViolet600,
+
+    // CheckBox
+    val checkBoxSelected: Color = ColorViolet500,
+    val checkBoxUnselected: Color = ColorNeutralWhite,
+    val checkBoxDisabled: Color = ColorGray200,
+
+    // Chip
+    val chipSelected: Color = ColorViolet100,
+    val chipUnselected: Color = ColorGray100,
+    val chipDisabled: Color = ColorGray50,
+
+    // Pagination / Brand
+    val paginationBrandPressed: Color = ColorViolet50,
+
+    // Pagination / Basic
+    val paginationBasicSelected: Color = ColorNeutralBlack,
+    val paginationBasicUnselected: Color = ColorGray500,
+
+    )
+
+val lightColorScheme = ColorScheme()
+ val darkColorScheme = ColorScheme()
+
+internal val LocalColorScheme = staticCompositionLocalOf { lightColorScheme }
+val LocalContentColor = staticCompositionLocalOf { lightColorScheme.textBasicPrimary }
```

Comment:
`compositionLocalOf`로 하면 Provide로 값을 변경 했을 때 .currunt로 읽는 부분만 리컴포지션 되지만
`staticCompositionLocalOf`로 하면 값을 바꾸었을 때 content 전체가 리컴포지션된다고 해요

자주 바뀌는 건 `compositionLocalOf`, 거의 바뀌지 않는 건 `staticCompositionLocalOf`로 하는 게 좋다고 하네요
```suggestion
val LocalContentColor = compositionLocalOf { lightColorScheme.textBasicPrimary }
```

Reply:
[참고](https://charlezz.com/?p=46403)